pkg/webhook/pod/mutating/mutating_function.go

```diff
@@ -0,0 +1,12 @@
+package mutating
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type mutatingPodFunc func(h *PodMutatingHandler, ctx context.Context, req admission.Request, pod *corev1.Pod) error
```

Comment:
IMO, make the function type as plugin is more flexible.  We cann add TODO comments to make the refactor later or add issue to track the issue. 

Add add more UT please.